### ISSUE TITLE
Bote: Nicer Error page (Trac #1367)

### ISSUE THREAD

**Issue by str4d** [i2p/i2p.i2p-bote#30]
When clicking on a mail in e.g. outbox in order to read it, after it has been moved by Bote to sent folder, you get an NPE.
A screen saying the mail could not be found and was likely moved would look nicer. with a link back to inbox.

HTTP ERROR 500

Problem accessing /i2pbote/showEmail.jsp. Reason:

    Server Error

Caused by:

java.lang.NullPointerException
	at i2p.bote.web.ShowAttachmentsTag.doTag(ShowAttachmentsTag.java:49)
	at i2p.bote.jsp.showEmail_jsp._jspx_meth_ib_005fshowAttachments_005f0(showEmail_jsp.java:1176)
	at i2p.bote.jsp.showEmail_jsp._jspx_meth_ib_005frequirePassword_005f0(showEmail_jsp.java:218)
	at i2p.bote.jsp.showEmail_jsp._jspService(showEmail_jsp.java:107)
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1496)
	at i2p.bote.web.CharsetFilter.doFilter(CharsetFilter.java:38)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1476)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:501)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:557)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:255)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
	at org.eclipse.jetty.server.Server.handle(Server.java:370)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)
	at org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:971)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:1033)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:644)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:696)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:53)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Migrated from https://trac.i2p2.de/ticket/1367
```json
{
    "status": "closed", 
    "changetime": "2015-02-16T20:21:16", 
    "description": "When clicking on a mail in e.g. outbox in order to read it, after it has been moved by Bote to sent folder, you get an NPE.\nA screen saying the mail could not be found and was likely moved would look nicer. with a link back to inbox.\n\nHTTP ERROR 500\n\nProblem accessing /i2pbote/showEmail.jsp. Reason:\n\n    Server Error\n\nCaused by:\n\njava.lang.NullPointerException\n\tat i2p.bote.web.ShowAttachmentsTag.doTag(ShowAttachmentsTag.java:49)\n\tat i2p.bote.jsp.showEmail_jsp._jspx_meth_ib_005fshowAttachments_005f0(showEmail_jsp.java:1176)\n\tat i2p.bote.jsp.showEmail_jsp._jspx_meth_ib_005frequirePassword_005f0(showEmail_jsp.java:218)\n\tat i2p.bote.jsp.showEmail_jsp._jspService(showEmail_jsp.java:107)\n\tat org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:848)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1496)\n\tat i2p.bote.web.CharsetFilter.doFilter(CharsetFilter.java:38)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1476)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:501)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:557)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:255)\n\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)\n\tat org.eclipse.jetty.server.Server.handle(Server.java:370)\n\tat org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)\n\tat org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:971)\n\tat org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:1033)\n\tat org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:644)\n\tat org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)\n\tat org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)\n\tat org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:696)\n\tat org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:53)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\n\tat java.lang.Thread.run(Unknown Source)\n", 
    "reporter": "user", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1424118076570797", 
    "component": "apps/plugins", 
    "summary": "Bote: Nicer Error page", 
    "priority": "trivial", 
    "keywords": "I2P-Bote error usability easy", 
    "version": "0.9.14.1", 
    "parents": "", 
    "time": "2014-08-28T09:06:16", 
    "milestone": "0.9.18", 
    "owner": "str4d", 
    "type": "defect"
}
```

**Comment by str4d** [comment]
Trac update at 20150109T07:14:56: str4d changed keywords from "Bote, error page" to "I2P-Bote error usability easy"

**Comment by str4d** [comment]
Trac update at 20150130T06:03:34: 
* str4d commented:

> Fixed in `e48223ee0efd58fd3bed2e2a425affc8d2f1f0d2`.


* str4d changed milestone from "" to "0.9.18"
* str4d changed resolution from "" to "fixed"
* str4d changed status from "new" to "closed"

**Comment by str4d** [comment]
Trac update at 20150216T20:21:16: user commented:

> thank you
